rs/cli/src/auth.rs

```diff
@@ -43,6 +47,162 @@ pub fn hsm_key_id_to_string(s: u8) -> String {
+        auth_opts: AuthOpts,
+        requirement: AuthRequirement,
+        network: &Network,
+        neuron_id: Option<u64>,
+    ) -> anyhow::Result<Self> {
+        match requirement {
+            AuthRequirement::Anonymous => Ok(Self {
+                auth: Auth::Anonymous,
+                neuron_id: 0,
+                include_proposer: false,
+            }),
+            AuthRequirement::Signer => Ok(Self {
+                auth: Auth::from_auth_opts(auth_opts).await?,
+                neuron_id: 0,
+                include_proposer: false,
+            }),
+            AuthRequirement::Neuron => Ok({
+                let auth = Auth::from_auth_opts(auth_opts).await?;
+                Self {
+                    neuron_id: match neuron_id {
+                        Some(n) => n,
+                        None => auth.auto_detect_neuron_id(network.nns_urls.clone()).await?,
+                    },
+                    auth,
+                    include_proposer: true,
+                }
+            }),
+            AuthRequirement::OverridableBy {
+                network: required_network,
+                neuron,
+            } => {
+                let maybe_user_provided_neuron = Auth::from_auth_opts(auth_opts).await.map(|auth| {
+                    (
+                        match neuron_id {
+                            Some(n) => {
+                                Box::pin(async move { Ok(n) }) as Pin<Box<dyn futures::Future<Output = std::result::Result<u64, anyhow::Error>>>>
+                            }
+                            None => Box::pin({
+                                let auth_clone = auth.clone();
+                                async move { auth_clone.auto_detect_neuron_id(network.nns_urls.clone()).await }
+                            }),
+                        },
+                        auth,
+                    )
+                });
+
+                let (neuron_id, auth) = if required_network.eq(network) {
+                    // The network matches, check if the override is possible
+                    let maybe_neuron_override = match neuron.auth {
+                        // Soft error. This will error only if
+                        // the user didn't provide any of his auth
+                        Auth::Keyfile { path } if !path.exists() => Err(anyhow::anyhow!(
+                            "Path `{}` not found, which can be used to override this command. Specify your own auth args",
+                            path.display()
+                        )),
+                        Auth::Keyfile { path } => Ok((neuron.neuron_id, Auth::Keyfile { path })),
+                        // Hard error on the whole program since we don't support this
+                        _ => anyhow::bail!("Overriding neuron with auth types other than keyfile is not supported"),
+                    };
+
+                    match (maybe_user_provided_neuron, maybe_neuron_override) {
```

Comment:
Do we still have a use case to preserve the override authentication branch?  What happens, hypothetically speaking, if we completely delete it?  What becomes impossible?

Reply:
I would love to just remove it. The use case seemed pretty good back then => If user provides nothing, assume OVERRIDE as default. Only example right now is to submit an `update_unassigned_nodes` with automation as default. Right now that part is done by Airflow and I was thinking that removing it would make the most sense because it would simplify the code, and we review proposals before submitting anyway.

Reply:
I have decided to just remove this code. It looks way simpler and more readable. I've seen that the use case we had for this is super narrow and is a one-off.

Reply:
That is grreat news!